File: 205. Isomorphic Strings/Solution.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.size()!=t.size()){
            return false;
        }
        unordered_map<char,char> st;
        unordered_map<char,char> ts;
        for(int i=0; i<s.size(); i++){
            char c1 = s[i];
            char c2 = t[i];
            if(st.count(c1)){
                if(st[c1]!=c2){
                    return false;
                }
            }else{
                st[c1] = c2;
            }
            if(ts.count(c2)){
                if(ts[c2] != c1){
                    return false;
                }
            }else{
                ts[c2] = c1;
            }
        }
        return true;
    }
};

File: 205. Isomorphic Strings/Solution.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        st = {}
        ts = {}
        for c1,c2 in zip(s,t):
            if c1 in st:
                if st[c1]!=c2:
                    return False
            else:
                st[c1] = c2
            if c2 in ts:
                if ts[c2]!=c1:
                    return False
            else:
                ts[c2] = c1
        return True
